shamil_latypov/logger.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <fstream>
+#include <iostream>
+#include <optional>
+#include <string>
+#include "graph.hpp"
+
+namespace uni_cource_cpp {
+
+class Logger {
+ public:
+  static Logger& prepare_logger() {
+    static Logger logger;
+    return logger;
+  }
+
+  void log(const std::string& text);
+
+  void set_file_output_name(const std::string& file_path);
```

Comment:
Так как стрим у нас - опциональное значение, в таком случае имеет смысл и указывать файл, как опциональное значение. Соответственно, если передаем `std::nullopt` - значит нужно удалить стрим.
"set_file_output_name" - указать внешнее имя файла. Корявое название. Более того, ты указываешь не имя файла, а путь к файлу.
```suggestion
  void set_output_file_path(const std::optional<std::string>& file_path);
```